src/index.ts

```diff
@@ -66,49 +99,183 @@ class ServerlessMagento {
+          };
+
+          await createActivatorFunctionArtifact(
+               this.serverless.service.provider.region,
+               this.handlerFolder,
+               this.ssmPrefix,
+               this.adminInterfaces[0].name,
+               this.adminInterfaces[0].app_url,
+          );
+
+          addActivatorFunctionRoleToResources(
+               this.serverless.service,
+               this.ssmPrefix
+          );
+
+          addActivatorFunctionToService(this.serverless.service, activatorConfig);
+     }
+
+     injectInterfaceContext = async () => {
+          this.serverless.cli.log(`Writing APIGateway context to SSM`);
 
-          this.serverless.service.getAllFunctions().forEach((functionName) => {
-               const functionObj = this.serverless.service.getFunction(functionName);
-               if (functionObj.environment == null) {
-                    functionObj.environment = {};
+          const apiGatewayKey = await this.provider
+          .request('CloudFormation', 'describeStackResources', {
+               StackName: this.provider.naming.getStackName(),
+          })
+          .then((resources: any) => {
+               const apiKeys = resources.StackResources
+               .filter((resource: any) => resource.ResourceType === 'AWS::ApiGateway::ApiKey')
+               .map((resource: any) => resource.PhysicalResourceId)
+
+               if (apiKeys.length > 1) {
+                    throw new Error('Multiple APIGateway keys not supported');
                }
+               const keyId = apiKeys.pop();
 
-               functionObj.environment.MAGENTO_ACCESS_TOKEN=registrationResponse.access_token;
-               functionObj.environment.MAGENTO_SERVICE_ID=registrationResponse.service_id;
+               return this.provider.request('APIGateway', 'getApiKey', {
+                    apiKey: keyId,
+                    includeValue: true,
+               });
+          })
+          .then((apiKeys: any) => {
+               return apiKeys.value;
           });
-     }
 
+          const apiGatewayBaseUrl = await this.provider
+          .request('CloudFormation', 'describeStackResources', {
+               StackName: this.provider.naming.getStackName(),
+          })
+          .then((resources: any) => {
+               const apiRestApis = resources.StackResources
+               .filter((resource: any) => resource.ResourceType === 'AWS::ApiGateway::RestApi')
+               .map((resource: any) => resource.PhysicalResourceId)
+
+               if (apiRestApis.length > 1) {
+                    throw new Error('Multiple APIGatewayRestApis not supported');
+               } else if (apiRestApis.length == 0) {
+                    throw new Error('Multiple APIGatewayRestApis');
+               }
+
+               const restApiId = apiRestApis.pop();
+               return `https://${restApiId}.execute-api.ap-southeast-2.amazonaws.com/${this.serverless.service.provider.stage}/`
```

Comment:
Need to swap this out to the region of the service instead of hardcoding southeast-2

Reply:
Good spot! Thanks.